gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "cf93d5f0ce00"
+      },
+      "outputs": [],
+      "source": [
+        "MODEL_NAME = \"gemini-1.5-flash\"  # Choice of model\n",
+        "generation_config = {\"temperature\": 0}\n",
+        "\n",
+        "model = GenerativeModel(\n",
+        "    model_name=MODEL_NAME,\n",
+        "    generation_config=generation_config,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "VPylX87eREVh"
+      },
+      "source": [
+        "## Notebook Tutorial"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oP_UflnpUVRL"
+      },
+      "source": [
+        "### Resume Structure and Prompt Initialization\n",
+        "\n",
+        "After installing necessary packages and loading the language model, we initialize predefined prompts designed for resume generation and the related structure. These prompts guide the model to extract relevant information from an existing resume and create a tailored output that aligns with the provided job description."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "1d17bfb0-f4ee-46b9-87b5-7b04ca6a1129"
+      },
+      "outputs": [],
+      "source": [
+        "\"\"\"\n",
+        "------------------------\n",
+        "EXTRACTION\n",
+        "------------------------\n",
+        "Personal Information\n",
+        "Education\n",
+        "Languages\n",
+        "Extracurricular_Activities\n",
+        "References\n",
+        "Experience\n",
+        "\n",
+        "------------------------\n",
+        "GENERATION\n",
+        "------------------------\n",
+        "Professional_Summary\n",
+        "\n",
+        "--------------------------------------------------\n",
+        "ARRANGEMENT, TRUNCATION, GENERATION\n",
+        "--------------------------------------------------\n",
+        "Coursework\n",
+        "Skills\n",
+        "Certifications\n",
+        "\n",
+        "\"\"\"\n",
+        "\n",
+        "PROMPT_EXTR = \"\"\"\n",
+        "You are a seasoned career advising professional expert in crafting resumes and cover letters.\n",
+        "\n",
+        "Consider the below details that have been extracted from a resume\n",
+        "%s\n",
+        "\n",
+        "INSTRUCTIONS:\n",
+        "- Maintain truthfulness and objectivity.\n",
+        "- Prefer active voice over passive voice.\n",
+        "- The output should have the following valid JSON format\n",
+        "{\n",
+        "\"Personal_Information\": {\"Name\": \"\", \"Address\": \"\", \"Mobile_No\": \"\", \"Email\": \"\"},\n",
+        "\"Education\": \"\",\n",
+        "\"Experience\": \"\",\n",
+        "\"Languages\": \"\",\n",
+        "\"Extracurricular_Activities\": \"\",\n",
+        "\"References\": \"\",\n",
+        "}\n",
```

Comment:
Instead of prompting for a specific JSON structure, use [Controlled Generation](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/control-generated-output)

Reply:
Replaced with GenerationConfig class and added controlled generation across the code